test/integration/fake_upstream.cc

```diff
@@ -629,7 +632,10 @@ void FakeUpstream::createUdpListenerFilterChain(Network::UdpListenerFilterManage
 }
 
 void FakeUpstream::threadRoutine() {
-  socket_factory_->doFinalPreWorkerInit();
+  if (!manual_listen_enabled_ && !listen_started_) {
+    socket_factory_->doFinalPreWorkerInit();
+    listen_started_ = true;
```

Comment:
What is the purpose of this extra variable?

Reply:
This variable is used to guard unintended effects here. https://github.com/envoyproxy/envoy/pull/19644/files#diff-04016f2a53109c31115f65ad41df56c4c237f343cf46a1f1fe4719a5cfc7b569R838